### PR TITLE
feat: upgrade to Nifi 1.27.0 and add compatibility with Java 17

### PR DESCRIPTION
**Pull request by ranim-n**

### CODE REVIEW

File: nifi-ngsi-bundle/nifi-ngsi-nar/pom.xml [hunk 1]
```diff
@@ -19,11 +19,11 @@
     <parent>
         <groupId>org.apache.nifi</groupId>
         <artifactId>nifi-ngsi-bundle</artifactId>
-        <version>1.3.6</version>
+        <version>2.1.0</version>
```
Comment (bobeal) [inline nifi-ngsi-bundle/nifi-ngsi-nar/pom.xml]: why 2.1.0?

File: nifi-ngsi-bundle/nifi-ngsi-processors/pom.xml [hunk 2]
```diff
@@ -87,7 +92,7 @@
         <dependency>
             <groupId>org.apache.nifi</groupId>
             <artifactId>nifi-processor-utils</artifactId>
-            <version>${nifi.version}</version>
+            <version>1.15.3</version>
```
Comment (bobeal) [inline nifi-ngsi-bundle/nifi-ngsi-processors/pom.xml]: why did you remove `${nifi.version}` (which avoid repeating the NiFi version everywhere) and replace it with an old version??

File: nifi-ngsi-bundle/nifi-ngsi-nar/pom.xml (hunk 1 above)
Comment (ranim-n) [reply nifi-ngsi-bundle/nifi-ngsi-nar/pom.xml]: it is the latest version in https://github.com/ging/fiware-draco/blob/a4662f421b88e6d68b82c68c73679ad4e29540f4/nifi-ngsi-bundle/nifi-ngsi-nar/pom.xml#L22

File: nifi-ngsi-bundle/nifi-ngsi-processors/pom.xml (hunk 2 above)
Comment (ranim-n) [reply nifi-ngsi-bundle/nifi-ngsi-processors/pom.xml]: The nifi version `1.27.0 ` doesn't match for this dependency, the latest version available is `1.15.3`

**Comment by bobeal** [comment]
rebased on the current master (was outdated here)

File: nifi-ngsi-bundle/nifi-ngsi-processors/pom.xml (hunk 2 above)
Comment (bobeal) [reply nifi-ngsi-bundle/nifi-ngsi-processors/pom.xml]: this is because it no longer exists after 1.15.3: https://issues.apache.org/jira/browse/NIFI-9610

so this dependency should be removed (and new ones added only if necessary).

File: nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/http/HttpBackend.java [hunk 3]
```diff
@@ -226,12 +231,12 @@ public Object run() {
                 Set<Principal> principals = current.getPrincipals();
 
                 for (Principal next : principals) {
-                    System.out.println("DOAS Principal: " + next.getName());
+                    logger.info("DOAS Principal: {}", next.getName());
                 } // for
```
Comment (thomasBousselin) [inline nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/http/HttpBackend.java:235]: something like logger.info(String.join(principals.map("DOAS Principal" + it.getName())))
could repace this logging loop.
